staging/src/k8s.io/client-go/scale/scheme/types.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheme
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// This file contains our own "internal" version of scale that we use for conversions,
+// since we can't use the main Kubernetes internal versions.
```

Comment:
This makes me very sad, but so be it.

Reply:
IIUC with the "alternative API representations", client can request the server to do the conversion. So the internal types and the entire conversion machinery can be removed in the future. Am I too optimistic?

Reply:
> IIUC with the "alternative API representations", client can request the server to do the conversion. So the internal types and the entire conversion machinery can be removed in the future. Am I too optimistic?

Too optimistic.  This code will still be correct even once we get there.  I would guess we're multiple releases away from re-loosening the server to support this.